src/Responsive.php

```diff
@@ -85,19 +86,53 @@ public static function getPresets(Asset $image): array
         }
 
         if (! isset($presets['placeholder'])) {
-            $glideUrl = Statamic::tag('glide:data_url')->params(['preset' => collect($configPresets)->keys()->first(), 'src' => $image->url(), 'fit' => 'crop_focal'])->fetch();
+            $glideUrl = Statamic::tag('glide:data_url')->params(['preset' => collect($configPresets)->keys()->first(), 'src' => $asset->url(), 'fit' => 'crop_focal'])->fetch();
             $presets['placeholder'] = $glideUrl;
         }
 
         return $presets;
     }
 
-    protected static function getPresetsByRatio(Asset $image, array $config): array
+    protected static function getGlideUrl(Asset $asset, string $preset, string $fit, ?string $format = null): ?string
+    {
+        if ($preset === 'placeholder') {
+            return Statamic::tag('glide:data_url')->params(['preset' => $preset, 'src' => $asset->url(), 'format' => $format, 'fit' => $fit])->fetch();
+        }
+
+        $manipulator = self::getManipulator($asset, $preset, $fit, $format);
+        $params = $manipulator->getParams();
+
+        $manipulationCacheKey = 'asset::' . $asset->id() . '::' . md5(json_encode($params));
+
+        if (Glide::cacheStore()->has($manipulationCacheKey)) {
+            $cachedUrl = Glide::cacheStore()->get($manipulationCacheKey);
```

Comment:
You could change the if to
```
        if ($cachedUrl = Glide::cacheStore()->get($manipulationCacheKey)) {
```

Because when it's empty it will return a false value.